### ISSUE TITLE
false logging-too-many-args with logging-format-style=fstr

### ISSUE THREAD

**Issue by belm0** [PyCQA/pylint#3362]
### Steps to reproduce
1. in pylintrc, set `logging-format-style=fstr` and `disable=logging-format-interpolation`
2. have logging statement in code using % formatting
```
logger.warning('hello %s %s', 'foo', 'bar')
```
### Current behavior
E1205: Too many arguments for logging format string (logging-too-many-args)

### Expected behavior
no error, since the code is valid

logging-format-style should have no bearing, since logging-format-interpolation is disabled

### pylint --version output
```
pylint 2.4.4
astroid 2.3.3
Python 3.7.5 (default, Nov  1 2019, 02:16:38)
[Clang 10.0.0 (clang-1000.11.45.5)]
```

**Comment by askpatrickw** [comment]
@belm0 and @PCManticore  I believe this has regressed, or is affected by Python 3.9. Would you mind trying it out as well.

This is line of code: `logger.info("Found device: %s", device_dir) # logs correctly, Pylint E1205`

```cli
pylint --version
pylint 2.6.0
astroid 2.4.2
Python 3.9.0 (default, Oct  9 2020, 12:02:34) 
[Clang 11.0.0 (clang-1100.0.33.17)]

pylint circup.py
************* Module circup
circup.py:275:4: E1205: Too many arguments for logging format string (logging-too-many-args)
```

**Comment by hippo91** [comment]
@askpatrickw i'm unable to reproduce this. For example linting the following code :
```
#pylint: disable=missing-module-docstring, invalid-name
import logging

logger = logging.getLogger("toto")
device_dir = "/dev/uart"

logger.info("Found device: %s", device_dir)
```
doesn't produce any warning.
```
pylint 2.6.0
astroid 2.4.3
Python 3.9.0 (default, Oct 10 2020, 11:00:00) 
[GCC 8.3.0]
```

**Comment by bdlb77** [comment]
@hippo91 I have the same error. 

```python
import logging
logging.basicConfig(level=logging.INFO)
log: logging.Logger = logging.getLogger(__name__)

log.info("Result File: %s", filename)
```
```
: E1205: Too many arguments for logging format string (logging-too-many-args)
```

**Comment by askpatrickw** [comment]
@hippo91  what OS, Python version were you running?

I am on MacOS, Python 3.9.

**Comment by hippo91** [comment]
@askpatrickw i'm wondering if it could be linked to the OS. Have you the possibility to test it on another OS?

**Comment by askpatrickw** [comment]
It also failed in this github actions run on Ubuntu 18.04.5
https://github.com/adafruit/circup/pull/41/checks?check_run_id=1300369856#step:12:7

**Comment by belm0** [comment]
everyone is neglecting to note their logging-related pylintrc configuration

for example, that circup case has `logging-format-style=new` in pylintrc, which seems to be related to the emitted error

I believe with `logging-format-style=new`, the checker is strictly expecting `{}` interpolation

**Comment by askpatrickw** [comment]
Thank you @belm0. I wasn't aware of that setting and didn't setup the original .pylintrc.
My PR It is good with that setting, again much thanks!

**Comment by hippo91** [comment]
@belmo thanks for noting this. You are absolutely right.
